cmd/graph.go

```diff
@@ -52,6 +58,83 @@ var graphCmd = &cobra.Command{
 	},
 }
 
+// find all possible chains starting from currentDep
+func getAllChains(currentDep string, graph map[string][]string, currentChain Chain, chains *[]Chain) {
+	currentChain = append(currentChain, currentDep)
+	_, ok := graph[currentDep]
+	if ok {
+		for _, dep := range graph[currentDep] {
+			if !contains(currentChain, dep) {
+				cpy := make(Chain, len(currentChain))
+				copy(cpy, currentChain)
+				getAllChains(dep, graph, cpy, chains)
+			} else {
+				*chains = append(*chains, currentChain)
+			}
+		}
+	} else {
+		*chains = append(*chains, currentChain)
+	}
+}
+
+// get the contents of the .dot file for the graph
+// when the -d flag is set
+func getFileContentsForSingleDep(chains []Chain, dep string) string {
+	data := ""
+	// to color the entered node as yellow
+	data += fmt.Sprintf("MainNode [label=\"%s\", style=\"filled\" color=\"yellow\"]", dep)
+
+	// add all chains which have the input dep to the .dot file
+	for _, chain := range chains {
+		if chainContains(chain, dep) {
+			data += "\n"
+			for i := range chain {
+				if chain[i] == dep {
+					chain[i] = "MainNode"
+				} else {
+					chain[i] = "\"" + chain[i] + "\""
+				}
+			}
+			data += strings.Join(chain, " -> ")
+		}
+	}
+	return data
+}
+
+// get the contents of the .dot file for the graph
+// of all dependencies (when -d is not set)
+func getFileContentsForAllDeps(deps []string, depGraph map[string][]string, mainModule string) string {
+	data := ""
+	// color the main module as yellow
+	data += fmt.Sprintf("MainNode [label=\"%s\", style=\"filled\" color=\"yellow\"]\n", mainModule)
```

Comment:
this is shared between the function, maybe we could extract it to func with dependency of module name